sw/device/tests/sim_dv/alert_handler_escalation.c

```diff
@@ -0,0 +1,162 @@
+}};
+
+/**
+ * External ISR.
+ *
+ * Handles all peripheral interrupts on Ibex. PLIC asserts an external interrupt
+ * line to the CPU, which results in a call to this OTTF ISR. This ISR
+ * overrides the default OTTF implementation.
+ */
+void ottf_external_nmi_handler(void) {
+  LOG_INFO("You are experiencing an NMI");
+
+  // Now intentionally hang the device
+  CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+      &clkmgr, kTopEarlgreyGateableClocksIoDiv4Peri, kDifToggleDisabled));
+
+  // access uart after clocks have been disabled
+  CHECK_DIF_OK(dif_uart_disable_rx_timeout(&uart));
+  LOG_FATAL("This message should never be seen");
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+
+  CHECK_DIF_OK(dif_uart_init(
+      mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart));
+
+  CHECK_DIF_OK(dif_keymgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_KEYMGR_BASE_ADDR), &keymgr));
+
+  // Check if there was a HW reset caused by the escalation.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  if (rst_info & kDifRstmgrResetInfoPor) {
+    // set the alert we care about to class A
+    CHECK_DIF_OK(dif_alert_handler_configure_alert(
+        &alert_handler, kTopEarlgreyAlertIdRvCoreIbexRecovSwErr,
+        kDifAlertHandlerClassA, /*enabled=*/kDifToggleEnabled,
+        /*locked=*/kDifToggleEnabled));
+
+    // configurate class A
+    CHECK_DIF_OK(dif_alert_handler_configure_class(
+        &alert_handler, kDifAlertHandlerClassA, kConfigProfiles[0],
+        /*enabled=*/kDifToggleEnabled,
+        /*locked=*/kDifToggleEnabled));
+
+    // Initialize keymgr with otp contents
+    keymgr_testutils_advance_state(&keymgr, NULL);
+
+    // DO NOT REMOVE, DV sync message
+    LOG_INFO("Keymgr entered Init State");
+
+    // Enable NMI
+    CHECK_DIF_OK(dif_rv_core_ibex_enable_nmi(&rv_core_ibex,
+                                             kDifRvCoreIbexNmiSourceAlert));
+
+    // force trigger the alert
+    CHECK_DIF_OK(dif_rv_core_ibex_alert_force(&rv_core_ibex,
+                                              kDifRvCoreIbexAlertRecovSwErr));
+
+    // Stop execution here and just wait for something to happen
+    wait_for_interrupt();
+    CHECK(false, "Should have reset before this line");
+
+  } else if (rst_info & kDifRstmgrResetInfoEscalation) {
+    LOG_INFO("We have reset for an amazing reason");
```

Comment:
Can we put more dry comment like
"We have a reset for ResetInfoEscalation."
this will save some time to debug this test failure without reading a test file.

Reply:
haha yeah, i left that there when i was debugging, i'll make sure to make it something more sensible

Reply:
updated